File: configlibrary/src/main/java/com/shmedo/configlibrary/ble/enums/SaveConfigMode.java
package com.shmedo.configlibrary.ble.enums;

/**
 * 项目名：  mCloudapp <br/>
 * 包名：    com.shmedo.core.enums <br/>
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/4/23 <br/>
 * 描述：    保存设备配置参数方式
 */
public enum SaveConfigMode {

    //保存并重启
    SAVE_REBOOT(1),

    //保存
    SAVE_NO_REBOOT(2);

    private int model;
    SaveConfigMode(int model) {
        this.model = model;
    }

    public int toInt() {
        return model;
    }

    public static SaveConfigMode valueOf(int model) {
        switch (model) {
            case 2:
                return SAVE_NO_REBOOT;

            default:
                return SAVE_REBOOT;
        }
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/md_cmd/enums/SaveConfigMode.kt
package com.shmedo.lib.device.base.md_cmd.enums

/**
 * 项目名：  mCloudapp <br></br>
 * 包名：    com.shmedo.core.enums <br></br>
 * 创建者:   gonghe <br></br>
 * 创建时间:  2020/4/23 <br></br>
 * 描述：    保存设备配置参数方式
 */
enum class SaveConfigMode(private val model: Int) {
    //保存并重启
    SAVE_REBOOT(1),

    //保存
    SAVE_NO_REBOOT(2);

    fun toInt(): Int {
        return model
    }

    companion object {
        fun value(model: Int): SaveConfigMode {
            return when (model) {
                2 -> SAVE_NO_REBOOT
                else -> SAVE_REBOOT
            }
        }
    }
}
